src/client/components/atoms/Button/Button.jsx

```diff
@@ -5,132 +5,110 @@ import clsx from 'clsx';
-    'btn__container--cols3': !isIcon && (icon && !iconCenter && !iconPosition),
-    'btn__container--cols3-right': !isIcon && (icon && !iconCenter && iconPosition),
+  const contentClassName = clsx({
+    'btn__content': true,
+    'btn__content--cols2': !isIcon && (icon && centered),
+    'btn__content--cols3': !isIcon && (icon && !centered && !iconRight),
+    'btn__content--cols3-right': !isIcon && (icon && !centered && iconRight),
   });
 
   return (
     <button
       onClick={onClick}
-      type={typeButtons[type]}
-      className={btnStyles}
+      type={type}
+      className={buttonClassName}
       disabled={disabled}
       form={form}
     >
-      { icon ? (
-        <div className={btnContainerStyles}>
-          {!iconPosition
-            && <div className='btn__container--icon'>{icon}</div>}
+      {icon ? (
+        <div className={contentClassName}>
+          {!iconRight && <div className='btn__container--icon'>{icon}</div>}
           <div className='btn__container--children'>{children}</div>
-          {iconPosition
-            && <div className='btn__container--icon'>{icon}</div>}
+          {iconRight && <div className='btn__container--icon'>{icon}</div>}
         </div>
-      ) : (
-        children
-      )}
+      ) : children}
     </button>
   );
 };
 
 Button.propTypes = {
   /** Function that will be called on click event. */
   onClick: PropTypes.func.isRequired,
-
   /** The text inside the button or another component */
   children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
-
   /** Specify the type of the button */
-  type: PropTypes.string,
-
+  type: PropTypes.oneOf(['button', 'submit', 'reset', 'icon']),
   /** Element Icon type */
   icon: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
-
   /** Spacing modes for icon type */
-  iconMode: PropTypes.string,
-
-  /** Indicates on which side of the text the icon is placed. left = false | Right = true */
-  iconPosition: PropTypes.bool,
-
+  iconMode: PropTypes.oneOf(iconModes),
+  /** Indicates is placed on the right side of the text */
+  iconRight: PropTypes.bool,
   /** Icon and other elements will be centered. */
-  iconCenter: PropTypes.bool,
-
+  centered: PropTypes.bool,
   /** Boolean class to specify if the color button is primary, secondary or warning */
-  color: PropTypes.string,
-
+  color: PropTypes.oneOf(colors),
   /** Class to overwrite the styles */
   className: PropTypes.string,
-
   /** Specify if the button is disabled or not */
   disabled: PropTypes.bool,
-
   /** If the button is outside a form and you want to active the submit event on click, you can pass the form id */
   form: PropTypes.string,
 };
 
 Button.defaultProps = {
   type: 'button',
-  icon: undefined,
-  iconMode: undefined,
-  iconPosition: false,
-  iconCenter: false,
-  color: 'default',
-  className: undefined,
+  icon: null,
+  iconMode: null,
+  iconRight: false,
+  centered: false,
+  color: 'primary',
+  className: '',
```

Comment:
Because it will always be a string, that is what the component is expecting to receive. Also it's a false value. So, it's great for both cases, it's what the component is expecting to receive and you can do, lets say: ```!className```